src/lambda/template.yaml

```diff
@@ -0,0 +1,101 @@
+AWSTemplateFormatVersion: '2010-09-09'
+
+Transform: AWS::Serverless-2016-10-31
+
+Description:  >
+  python3.8
+
+  SAM Template for sns-to-glue
+
+Parameters:
+
+  WorkflowName:
+    Type: String
+    Description: Name of the glue workflow to start when new data arrives
+
+  SsmParameterName:
+    Type: String
+    Description: >-
+      Name of the ssm parameter used by the lambda to authenticate with Synapse
+
+Resources:
+
+  SnsToGlueRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - lambda.amazonaws.com
+          Action:
+          - sts:AssumeRole
+      Policies:
+      - PolicyName: GetSsmParam
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - ssm:GetParameter
+            Resource:
+            - !Sub arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${SsmParameterName}
+      - PolicyName: StartGlueWorkflow
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - glue:StartWorkflowRun
+            - glue:PutWorkflowRunProperties
+            Resource:
+            - !Sub arn:aws:glue:${AWS::Region}:${AWS::AccountId}:workflow/${WorkflowName}
+
+  SnsToGlueFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      PackageType: Image
+      Role:
+        Fn::GetAtt:
+        - SnsToGlueRole
+        - Arn
+      Events:
+        SnsToGlue:
+          Type: SNS
+          Properties:
+            Topic: !Ref SnsTopic
+      Environment:
+        Variables:
+          SSM_PARAMETER_NAME:
+            Ref: SsmParameterName
+          GLUE_WORKFLOW_NAME:
+            Ref: WorkflowName
+      Timeout: 60
+      ImageUri: snstogluefunction:python3.8-v1
```

Comment:
Is this image auto generated?

Reply:
`sam build` builds the images for the lambda and updates the template. Actually, this line is relic of when I had to go recover the template from `.aws_sam/build` dir -- I think I'll remove it to avoid confusion.

Reply:
And to be more clear, when the image is built it is this block that determines how the image gets labeled:
```
    Metadata:
      Dockerfile: Dockerfile
      DockerContext: ./sns_to_glue
      DockerTag: v0.1
```